### PR TITLE
[T2A2][T16-A2]ChingHuiQi

### PR DESCRIPTION
**Pull request by HQching**

### CODE REVIEW

**Comment by HQching** [comment]
Ready for review

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -18,43 +18,39 @@
...
 import java.util.HashSet;
 import java.util.Optional;
 import java.util.Scanner;
 import java.util.Set;
 
 /*
- * NOTE : =============================================================
- * This class header comment below is brief because details of how to
- * use this class are documented elsewhere.
+ * NOTE : ============================================================= This
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:28]: errr! i guess this is <kbd>Ctrl</kbd> <kbd>Shift</kbd> <kbd> f </kbd> effect?
need to use it judiciously :-)

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -74,11 +70,11 @@
     private static final String MESSAGE_DISPLAY_PERSON_DATA = "%1$s  Phone Number: %2$s  Email: %3$s";
     private static final String MESSAGE_DISPLAY_LIST_ELEMENT_INDEX = "%1$d. ";
     private static final String MESSAGE_GOODBYE = "Exiting Address Book... Good bye!";
-    private static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format: %1$s " + LS + "%2$s";
+    private static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format: %1$s " + LINE_SEPARATOR
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:73]: good catch!

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -169,39 +154,45 @@
...
+     * concepts such as interface inheritance.
+     * =========================================================================
+     * ===========================
      */
 
     /**
      * List of all persons in the address book.
      */
-    private static final ArrayList<String[]> ALL_PERSONS = new ArrayList<>();
+    private static final ArrayList<HashMap<PersonProperty, String>> ALL_PERSONS = new ArrayList<>();
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:172]: you can explicitly set the type of the arraylist returned here.

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -210,35 +201,35 @@ public static void main(String[] args) {
         showWelcomeMessage();
         processProgramArgs(args);
         loadDataFromStorage();
+        processUserInput();
+    }
+
+    private static void processUserInput() {
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:207]: well `processUserInput` gives the impression that this is a one-time process. however, the process is in a loop repeats until the user chooses to exit. 
in light of this behavior, you may want to choose a better suited name

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -266,15 +257,14 @@ private static void processProgramArgs(String[] args) {
             setupGivenFileForStorage(args[0]);
         }
 
-        if(args.length == 0) {
+        if (args.length == 0) {
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:260]: :+1:  good attention to detail

File: src/seedu/addressbook/AddressBook.java [hunk 6]
```diff
@@ -391,15 +378,20 @@ private static String executeCommand(String userInputString) {
...
      *
-     * @return  size 2 array; first element is the command type and second element is the arguments string
+     * @return size 2 array; first element is the command type and second
+     *         element is the arguments string
      */
     private static String[] splitCommandWordAndArgs(String rawUserInput) {
-        final String[] split =  rawUserInput.trim().split("\\s+", 2);
-        return split.length == 2 ? split : new String[] { split[0] , "" }; // else case: no parameters
+        final String[] split = rawUserInput.trim().split("\\s+", 2);
+        return split.length == 2 ? split : new String[] { split[0], "" }; // else
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:386]: hmmm notice that 2 and 0 are magic numbers here, can you do something to eliminate them?

File: src/seedu/addressbook/AddressBook.java [hunk 7]
```diff
@@ -770,66 +778,58 @@ private static void savePersonsToFile(ArrayList<String[]> persons, String filePa
...
+     * ======= INTERNAL ADDRESS BOOK DATA METHODS
+     * =========================================================================
+     * =======
      */
 
     /**
      * Adds a person to the address book. Saves changes to storage file.
      *
-     * @param person to add
+     * @param personToAdd to add
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:791]: good choice of variable name :+1:

File: src/seedu/addressbook/AddressBook.java [hunk 8]
```diff
@@ -770,66 +778,58 @@ private static void savePersonsToFile(ArrayList<String[]> persons, String filePa
...
+     * @param targetInModel the actual person inside the address book (exactPerson
+     *            == the person to delete in the full list)
      * @return true if the given person was found and deleted in the model
      */
-    private static boolean deletePersonFromAddressBook(String[] exactPerson) {
-        final boolean changed = ALL_PERSONS.remove(exactPerson);
-        if (changed) {
+    private static boolean hasDeletePersonFromAddressBook(HashMap<PersonProperty, String> targetInModel) {
+        final boolean isChanged = ALL_PERSONS.remove(targetInModel);
+        if (isChanged) {
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:808]: again, good work here :-)

File: src/seedu/addressbook/AddressBook.java [hunk 9]
```diff
@@ -959,12 +959,12 @@ private static boolean isPersonDataExtractableFrom(String personData) {
         final String[] splitArgs = personData.trim().split(matchAnyPersonDataPrefix);
         return splitArgs.length == 3 // 3 arguments
                 && !splitArgs[0].isEmpty() // non-empty arguments
-                && !splitArgs[1].isEmpty()
-                && !splitArgs[2].isEmpty();
+                && !splitArgs[1].isEmpty() && !splitArgs[2].isEmpty();
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:962]: consider removing these magic numbers

File: src/seedu/addressbook/AddressBook.java [hunk 10]
```diff
@@ -1136,26 +1127,24 @@ private static String getUsageInfoForExitCommand() {
...
     /*
-     * ============================
-     *         UTILITY METHODS
-     * ============================
+     * ============================ UTILITY METHODS ============================
      */
 
     /**
-     * Removes sign(p/, d/, etc) from parameter string
+     * Removes prefix(p/, d/, etc) from parameter string
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:1135]: the comment here still refers to addressbook, no? (think where else in general would you use `p/ , d/` ?
when you update the behavior of the method, please ensure you update the document to reflect the behavior.

in this particular case, intention was to make remove prefix method as generic as possible, definitely that should be reflected in the header comments

**Review by okkhoy — COMMENTED** [review]
nice to see you have followed the good practise of making numerous small commits. quite a good job on identifying code improvements as well. :+1:

**Comment by okkhoy** [comment]
@HQching some comments added. please take a look, ack and close this PR.
please keep up the good work.

on an unrelated note, please upload a photo of yours to github profile so that it helps in identification

**Comment by HQching** [comment]
Thank you for the review!

**Pull request closed without merging**